### PR TITLE
Allow to configure yt-dlp `--format-sort` argument

### PR DESCRIPTION
**Pull request by dsander**
This exposes the [`--format-sort`][1] yt-dlp option to the user.

Implements parts of #316

[1]: https://github.com/yt-dlp/yt-dlp#sorting-formats

### CODE REVIEW

**Comment by bbilly1** [comment]
Excellent, thanks for taking this on! :-)

You can register a default value in the config template [here](https://github.com/tubearchivist/tubearchivist/blob/testing/tubearchivist/home/config.json), then there is a startup management function that will load that into redis [here](https://github.com/tubearchivist/tubearchivist/blob/testing/tubearchivist/config/management/commands/ta_startup.py#L48-L57). This way we can assume everybody has that key set in their config, even if it's just set to `False` and we don't have to worry about any KeyErrors or such.

Then it might be worth it to mention in the markup on the settings page that no all codecs are compatible with all browsers. Especially Safari is very picky. Just in case people get confused why their video doesn't have audio anymore.

File: tubearchivist/home/src/download/yt_dlp_handler.py [hunk 1]
```diff
@@ -312,6 +312,8 @@ def _build_obs_user(self):
         """build user customized options"""
         if self.config["downloads"]["format"]:
             self.obs["format"] = self.config["downloads"]["format"]
+        if self.config["downloads"]["format_sort"]:
+            self.obs["format_sort"] = self.config["downloads"]["format_sort"].split(",")
```
Comment (bbilly1) [inline tubearchivist/home/src/download/yt_dlp_handler.py:316]: I think yt-dlp is picky here, I think they expect no spaces after the commas. But we humans have that kind of ingrained. Would be a nice quality of life if we can handle both here, something like:
```python
format_sort_list = [i.strip() for i in format_sort.split(",")]
```

Comment (dsander) [reply tubearchivist/home/src/download/yt_dlp_handler.py]: Good point, done. I hope the style is OK.

**Comment by dsander** [comment]
> You can register a default value in the config template [here](https://github.com/tubearchivist/tubearchivist/blob/testing/tubearchivist/home/config.json), then there is a startup management function that will load that into redis [here](https://github.com/tubearchivist/tubearchivist/blob/testing/tubearchivist/config/management/commands/ta_startup.py#L48-L57). This way we can assume everybody has that key set in their config, even if it's just set to `False` and we don't have to worry about any KeyErrors or such.

Thanks for the hint, I missed the default config when grepping through the code base to figure out where this change might need to be made 😄

**Pull request merged**